message/infrastructure/http/subscriber.go

```diff
@@ -1,16 +1,45 @@
 package http
 
 import (
+	"encoding/json"
+	"io/ioutil"
 	"net/http"
+	"strings"
 	"sync"
 
+	"github.com/pkg/errors"
+
 	"github.com/ThreeDotsLabs/watermill"
 	"github.com/ThreeDotsLabs/watermill/message"
 	"github.com/go-chi/chi"
 )
 
 type UnmarshalMessageFunc func(topic string, request *http.Request) (*message.Message, error)
 
+// DefaultUnmarshalMessageFunc retrieves the UUID and Metadata from request headers,
+// as encoded by DefaultMarshalMessageFunc.
+func DefaultUnmarshalMessageFunc(topic string, req *http.Request) (*message.Message, error) {
```

Comment:
also, some unit tests for DefaultMarshalMessageFunc will be great

Reply:
28107de858177132c98e27adc2100338411e459c, updated in b97952c46ffe87d5349ecb9ef5a191baa72fbd80 (changed signature)